patches/server/1047-Add-UUID-attribute-modifier-API.patch

```diff
@@ -4,8 +4,32 @@ Date: Thu, 9 Nov 2023 20:34:44 -0500
 Subject: [PATCH] Add UUID attribute modifier API
 
 
+diff --git a/src/main/java/io/papermc/paper/attribute/UnmodifiableAttributeInstance.java b/src/main/java/io/papermc/paper/attribute/UnmodifiableAttributeInstance.java
+index 52439f4b959c74027eb191a3af960c70beb978e8..a2c057d92ea34368c7efc538b6e5b15ef342c54e 100644
+--- a/src/main/java/io/papermc/paper/attribute/UnmodifiableAttributeInstance.java
++++ b/src/main/java/io/papermc/paper/attribute/UnmodifiableAttributeInstance.java
+@@ -5,6 +5,7 @@ import org.bukkit.attribute.Attribute;
+ import org.bukkit.attribute.AttributeModifier;
+ import org.bukkit.craftbukkit.attribute.CraftAttributeInstance;
+ 
++import java.util.UUID;
+ import java.util.Collection;
+ 
+ public class UnmodifiableAttributeInstance extends CraftAttributeInstance {
+@@ -18,6 +19,11 @@ public class UnmodifiableAttributeInstance extends CraftAttributeInstance {
+         throw new UnsupportedOperationException("Cannot modify default attributes");
+     }
+ 
++    @Override
++    public void removeModifier(UUID uuid) {
++        throw new UnsupportedOperationException("Cannot modify default attributes");
++    }
++
+     @Override
+     public void addModifier(AttributeModifier modifier) {
+         throw new UnsupportedOperationException("Cannot modify default attributes");
```

Comment:
I added the diff to `UnmodifiableAttributeInstance` within this patch instead of the original [default attributes patch](https://github.com/TonytheMacaroni/Paper/blob/master/patches/server/0659-Get-entity-default-attributes.patch#L9-L44) to match the [transient modifiers patch](https://github.com/TonytheMacaroni/Paper/blob/master/patches/server/0943-Add-transient-modifier-API.patch#L7-L22) - should I move it to the original patch?

Reply:
No, I think it is fine to add here. We do a similar thing with another method we add to AttributeInstance.